src/Umbraco.Cloud.StorageProviders.AzureBlob/AzureBlobComposer.cs

```diff
@@ -1,35 +1,45 @@
 using System;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
 using Umbraco.Cms.Core.Composing;
 using Umbraco.Cms.Core.DependencyInjection;
 using Umbraco.Cms.Web.Common.ApplicationBuilder;
-using Umbraco.StorageProviders.AzureBlob;
 
 namespace Umbraco.Cloud.StorageProviders.AzureBlob
 {
+    /// <summary>
+    /// Automatically configures Azure Blob Storage for use on Umbraco Cloud.
+    /// </summary>
+    /// <seealso cref="Umbraco.Cms.Core.Composing.IComposer" />
     public class AzureBlobComposer : IComposer
     {
+        /// <inheritdoc />
         public void Compose(IUmbracoBuilder builder)
         {
             if (builder == null) throw new ArgumentNullException(nameof(builder));
 
-            if (string.IsNullOrEmpty(Environment.GetEnvironmentVariable("UMBRACO__CLOUD__ENVIRONMENT"))) return;
-
-            builder.AddAzureBlobMediaFileSystem();
+            // Do a quick check to see if we're running on Umbraco Cloud
+            var environment = Environment.GetEnvironmentVariable("UMBRACO__CLOUD__ENVIRONMENT");
+            if (string.IsNullOrEmpty(environment)) return;
```

Comment:
The environment variable we need to check has changed, it's now `UMBRACO__CLOUD__ISRUNNINGONCLOUD` wich should have the value `true`

```suggestion
            var isRunningOnCloud = Environment.GetEnvironmentVariable("UMBRACO__CLOUD__ISRUNNINGONCLOUD");
            if (isRunningOnCloud != "true") return;
```

Reply:
That's an easy fix, although I would argue we should ensure it's compared using `string.Equals(isRunningOnCloud, "true", StringComparison.InvariantCultureIgnoreCase)` or parsed as a boolean using `bool.TryParse()`. Would be very confusing if it stops working when the value is set to `True` 😉 

It this checked before creating the `ConfigurationBuilder` to prevent a possible expensive operation? Otherwise we might as well use `configuration.GetValue<bool>("IsRunningOnCloud")`, as the configuration only looks at environment variables anyway...

Reply:
Either way is fine, but I don't expect it to change as it's part of our provisioning.

The reason for checking the environment instead of using the configuration object is because we don't wan't to make it to easy to enable it locally by overriding the setting, and we'll ever only set the environment variable on Cloud

Reply:
Directly using `Environment` or the manually created configuration shouldn't matter in that case, right? In any case it's possible to add the environment variables in your `launchSettings.json`, so it's just an additional sanity check/enable flag, right?

Reply:
Updated to get the setting using `configuration.GetValue<bool>("IsRunningOnCloud")`, as that makes it cleaner and we always need it when running on Umbraco Cloud anyway.

Reply:
I still think we should keep the env. variable check to make it explicit that it's the variable is required for it to run, it also makes it safer so we don't accidentally allow anyone to overwrite it on Cloud if we add another config source at one point

Reply:
I still don't understand why we want this. If the ~3 appsettings (required by the Umbraco Cloud specific configuration) is there then we enable it, if one or all of the required settings or not there than we don't enable it. I don't see a good reason for having yet another check of an arbitrary appsetting is needed, as if we want to hide or make it more difficult to use the configuration outside of Cloud.

Reply:
Sure we can remove it, it was only added because it was agreed at one point we should have this check in our Cloud assemblies to ensure we only loaded parts that were needed on Cloud when running on Cloud

Reply:
I think that was specific to licensed products/plugins

Reply:
@ronaldbarendse you can just remove the check then 👍